controllers/v1/observationsController.js

```diff
@@ -631,128 +630,140 @@ module.exports = class Observations extends Abstract {
+                            entitiesData,
+                            result.entities
+                        );
+
+                        response["message"] = messageData;
+
+                        response.result.push({
+                            "data" : data,
+                            "count" : 1
+                        });
 
-                                if( entitiesData[0].metaInformation.districtName ) {
-                                    entitiesData[0]["districtName"] = entitiesData[0].metaInformation.districtName;
+                    } else {
+                        response.result = [];
+                        let subEntitiesMatchingType = [];
+                        let parentId = [];
+                        parentId.push(req.query.parentEntityId );
+                        let subEntities = await subEntitiesWithMatchingType( parentId,result.entityType,subEntitiesMatchingType )
+               
+                        if( !subEntities.length > 0 ) {
+                            return resolve(subEntities) 
+                        }
+                        
+                        
+                        let searchText= req.searchText;
+                        if( searchText !== "" ){
+                            let matchEntities = [];
+                            subEntities.map( entityData => {
+                                if( entityData.name.match(new RegExp(searchText, 'i')) || entityData.code.match(new RegExp("^" + searchText, 'm')) ) {
+                                    matchEntities.push(entityData)
                                 }
-
-                                entitiesData[0] = _.pick(
-                                    entitiesData[0],
-                                    ["_id","name","externalId","addressLine1","addressLine2","districtName"]
-                                )
-                            }
-
-                            let data = 
+                            });
+                            subEntities = [];
+                            subEntities = matchEntities;
+                        }
+        
+                        let totalcount = subEntities.length;
+                        if (subEntities.length > 0) {
+                            let startIndex = req.pageSize * (req.pageNo - 1);
+                            let endIndex = startIndex + req.pageSize;
+                            subEntities = subEntities.slice(startIndex, endIndex);
+                        }
+                    
+                        let data = 
                             await entitiesHelper.observationSearchEntitiesResponse(
-                                entitiesData,
+                                subEntities,
                                 result.entities
-                            );
+                        )
+                        
 
-                            response["message"] = messageData;
+                        response["message"] = messageData;
```

Comment:
@VISHNUDAS-tunerlabs assign the message from apiresponse file, please don't create new variable